src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -114,8 +121,23 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 					this.updateResult(e.resultSet, e.rows);
 				}
 			}));
+			if (this._cellModel.executionState === CellExecutionState.Running || !this._incrementalGridRenderingEnabled) {
+				await this.renderGrid();
+			} else {
+				this.setLoading(true);
+				setTimeout(async () => {
+					await this.renderGrid();
+					this.setLoading(false);
+				});
+			}
+		}
+	}
+
+	setLoading(isLoading: boolean): void {
+		this.isLoading = isLoading;
+		if (!(this._changeRef['destroyed'])) {
```

Comment:
do you know when will this happen? sometimes this means we are not disposing objects properly.

Reply:
I wouldn't expect it to happen normally - but since the grid loading is being done asynchronously then I would imagine it could happen if the user closed the notebook before it finished rendering the grids.

Doing something like clearing the timeout when the grid is disposed might be a "better" way to fix this.

Reply:
Heres the previous issue in notebook.component.ts: https://github.com/microsoft/azuredatastudio/issues/4180